docs/roadmap.md

```diff
@@ -0,0 +1,178 @@
+2. Getting enough configuration options working to allow for basic operation
+3. Integrating the pingora library and getting basic reverse proxy operation working
+4. Start setting up build and release infrastructure
+5. Start working on observability, including structured logging
+
+For more information: https://github.com/memorysafety/river/blob/main/docs/release-notes/2024-04-29-v0.2.0.md
+
+## In Progress Milestones
+
+### "Spike 2.1" / v0.3.x
+
+This is the first part of work intended to take place during June and July 2024.
+
+This work is focused on "load balancing" use cases, including:
+
+1. Supporting Load Balancing of upstream servers
+2. Supporting Health Checks of upstream servers
+3. Supporting Service Discovery of upstream servers
+
+This work is in progress, and is wrapping up shortly.
+
+### "Spike 2.2" / v0.4.x
+
+This is the second part of work intended to take place during June and July 2024.
+
+This work is focused on "Developer and Operator Quality of Life" features, including:
+
+1. Supporting basic static HTML file serving
+2. Supporting semi-dynamic observability endpoints, e.g. for Prometheus polling
+3. Support for hot-reloading of configuration
+4. CI for build and test checks on pull requests
+
+### "Spike 2.3" / v0.5.x
+
+This is the third and final part of work intended to take place during June and July 2024.
+
+This work is focused on "initial Robustness" features, including:
+
+1. Rate limiting of connections and/or requests
+2. CIDR/API range-based filtering for rejecting connections
+
+## Future Milestones - towards 1.0
+
+The following milestones are working towards the requirements specified in the design document
+for `river`: https://github.com/memorysafety/river/blob/main/docs/what-is-it.md
+
+These milestones are the currently planned way of structuring major features in the approach
+towards a stable 1.0 release.
+
+### "ACME features" / v0.6.x
+
+This future work is focused on implementing ACME protocol support, to enable automatically obtaining
+and/or renewing TLS certificates from providers such as Lets Encrypt. This feature is expected to
+work without active human interaction.
+
+Requirements for this stage include:
+
+1. The application MUST support the use of the Automatic Certificate Management Environment (ACME)
+   protocol to obtain new TLS certificates.
+2. The application MUST support the use of ACME protocol to renew TLS certificates.
+3. The application MUST support the configuration of domain names to be managed (including obtaining
+   and renewal steps) automatically
+4. The application MUST support both fully qualified and wildcard domains.
+5. The application MUST support configuration of certificate renewal interval, from either:
+    1. The number of days since the certificate was acquired
+    2. The number of days until the certificate will expired
+6. The application MUST support API Version 2 of the ACME protocol
```

Comment:
I think this should say RFC 8555 ACME.  "version 1" / "version 2" are let's encrypt specific names.  "version 1" doesn't exist anymore, effectively